content/oci/oci.go

```diff
@@ -454,6 +455,76 @@ func (s *Store) writeIndexFile() error {
 	return os.WriteFile(s.indexPath, indexJSON, 0666)
 }
 
+// reloadIndex reloads the index and updates metadata by creating a new store.
+func (s *Store) reloadIndex(ctx context.Context) error {
+	newStore, err := NewWithContext(ctx, s.root)
+	if err != nil {
+		return err
+	}
+	s.index = newStore.index
+	s.storage = newStore.storage
+	s.tagResolver = newStore.tagResolver
+	s.graph = newStore.graph
+	return nil
+}
+
+// GC removes garbage from Store. The garbage to be cleaned are:
+//   - unreferenced (dangling) blobs in Store which have no predecessors
+//   - garbage blobs in the storage whose metadata is not stored in Store
+func (s *Store) GC(ctx context.Context) error {
+	s.sync.Lock()
+	defer s.sync.Unlock()
+
+	// get reachable nodes by reloading the index
+	err := s.reloadIndex(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to reload index: %w", err)
+	}
+	reachableNodes := s.graph.DigestSet()
+
+	// clean up garbage blobs in the storage
+	rootpath := filepath.Join(s.root, "blobs")
+	algDirs, err := os.ReadDir(rootpath)
+	if err != nil {
+		return err
+	}
+	for _, algDir := range algDirs {
+		if !algDir.IsDir() {
+			continue
+		}
+		alg := algDir.Name()
+		// skip unsupported directories
+		if !isKnownAlgorithm(alg) {
+			continue
+		}
+		algPath := path.Join(rootpath, alg)
+		dgstDirs, err := os.ReadDir(algPath)
```

Comment:
nit: rename `dgstDirs` to `digestEntries` for more meaningful names.

Reply:
renamed.